dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:meta/meta.dart';
+import 'package:stack_trace/stack_trace.dart';
+
+import 'protocol/noop_origin.dart'
+    if (dart.library.html) 'protocol/origin.dart';
+import 'protocol/sentry_stack_frame.dart';
+import 'sentry_options.dart';
+
+/// converts [StackTrace] to [SentryStackFrames]
+class SentryStackTraceFactory {
+  /// A list of string prefixes of module names that do not belong to the app, but rather third-party
+  /// packages. Modules considered not to be part of the app will be hidden from stack traces by
+  /// default.
+  final List<String> _inAppExcludes;
+
+  /// A list of string prefixes of module names that belong to the app. This option takes precedence
+  /// over inAppExcludes.
+  final List<String> _inAppIncludes;
+
+  SentryStackTraceFactory(SentryOptions options)
+      : _inAppExcludes = options?.inAppExcludes,
+        _inAppIncludes = options?.inAppIncludes;
```

Comment:
options should be mandatory, why do we use `?`

Reply:
as Dart is not non-nullable for now, options can be passed null

Reply:
thats why we do `Objects.requireNonNull` on java and throws if null, we'd do the same in Dart, this would mean a bug on the SDK and we should fix it.

Reply:
✅  I added non nullability verifications